docs/src/lucet-runtime/states.dot

```diff
@@ -0,0 +1,48 @@
+digraph G {
```

Comment:
Thanks so much for adding a visual to the docs.  I'm a very very visual person, so pictures always help me immensely.  There's something that seems not quite right about the picture.  States A - D all have a single domain and a single possible value for `terminable`.  State E stands apart in that it can have three possible Domain values and both possible values for `terminable`.  This may very well represent why KillSwitch was so challenging and even partially motivate the rationale for writing these beautiful docs.  That said, I wonder whether more states could be added to make the diagram less muddled?  I don't know the answer to this question.  It may very well make things worse.  😊

Reply:
My approach was to write states with the diagram showing permissible values in `KillState`, for the state being described. State E is really the catch-all of "what lucet_runtime sees right after `swap_and_run` returns". It can be reached from a hostcall exit (where Domain is `Hostcall`), from a guest returning (`Domain::Guest`), from running a cancelled guest (`Domain::Cancelled`), but really the `KillState` when reaching E no longer matters. The instance that any KillSwitch might terminate has just finished running, so they're no longer effectual, and we should be about to replace `kill_state` with a fresh new `KillState` for the next run anyway. This is the same as `Instance::run returns` over in `execution`'s [module-level docs](https://docs.rs/lucet-runtime-internals/0.6.1/lucet_runtime_internals/instance/execution/index.html), which is similarly weird in having three `execution_domain` states.

I was initially sketching this out with `execution_domain` and `terminable` as both `N/A` for state E, walked that back for being imprecise, but I'll switch back to `N/A` - that's more appropriate for readers here. Maintainer-facing documentation of what permissible states _should_ be is already in the module-level commentary, and we don't need to duplicate that.

Reply:
Agreed.  Thanks.